### PR TITLE
Code Review - Geoff - Oct 10, 2017

### PR DESCRIPTION
**Pull request by gaelder**
This is a pull request to serve as a code review and is not to be merged. There isn't any commit history because it all got left behind when I switched everything from pathways-search repo over to app-ui.

### CODE REVIEW

File: src/client/features/view/components/sidebar/index.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+const React = require('react');
```
Comment (d2fong) [inline src/client/features/view/components/sidebar/index.js]: to review

File: src/client/features/view/components/sidebar/menus/components/downloadOption.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+const React = require('react');
```
Comment (d2fong) [inline src/client/features/view/components/sidebar/menus/components/downloadOption.js]: to review

File: src/client/features/view/cy/events/click.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const bindClick = (cy) => {
```
Comment (d2fong) [inline src/client/features/view/cy/events/click.js]: to review: tippy events, etc.

File: src/client/features/view/cy/events/move.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const bindMove = (cy) => {
```
Comment (d2fong) [inline src/client/features/view/cy/events/move.js]: to review, seems like a prototype/skeleton implementation though.

File: src/client/features/view/index.js [hunk 5]
```diff
@@ -1,9 +1,164 @@
...
+const make_cytoscape = require('./cy/');
+const test = require('./example.js');
+
+const queryString = require('query-string');
+const PathwayCommonsService = require('../../services/index.js').PathwayCommonsService;
+// const io = require('socket.io-client');
+const CDC = require('../../services/index.js').CDC;
+
 class View extends React.Component {
+  constructor(props) {
```
Comment (d2fong) [inline src/client/features/view/index.js]: To review: rethink db client functionality in the View component

File: src/client/services/cdc/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const io = require('socket.io-client');
```
Comment (d2fong) [inline src/client/services/cdc/index.js]: to review: rethink db / socket code

File: src/server/database/accessDB.js [hunk 7]
```diff
@@ -0,0 +1,229 @@
+
```
Comment (d2fong) [inline src/server/database/accessDB.js]: to review: rethinkdb

File: src/server/database/auth.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+//Replace Authentication Code Here with User Login System Later
```
Comment (d2fong) [inline src/server/database/auth.js]: to review: rethink db stuff

File: src/server/database/buildDB.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (d2fong) [inline src/server/database/buildDB.js]: to review: db population

File: src/server/database/createDatabase.js [hunk 10]
```diff
@@ -0,0 +1,84 @@
+
```
Comment (d2fong) [inline src/server/database/createDatabase.js]: review: db

File: src/server/database/heuristics.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+// --------- Fake Heuristics ---------------
```
Comment (d2fong) [inline src/server/database/heuristics.js]: review: db

File: src/styles/general.css [hunk 12]
```diff
@@ -0,0 +1,23 @@
+.flexCenter {
```
Comment (d2fong) [inline src/styles/general.css]: review: new general css added

File: src/styles/features/view.css [hunk 13]
```diff
@@ -1,3 +1,293 @@
-.view {
```
Comment (d2fong) [inline src/styles/features/view.css]: review: view css

**Comment by jonahdlin** [comment]
Commit on October 12 around 12:30 has the most recent frontend code. I would appreciate a code review of this as well.

File: src/client/features/view/index.js [hunk 14]
```diff
@@ -1,9 +1,182 @@
...
+    PathwayCommonsService.query(query.uri, 'json', 'Entity/dataSource/displayName')
+      .then(responseObj => {
+        var dsStr = responseObj ? responseObj.traverseEntry[0].value.pop() : '';
+        dsStr = dsStr ? dsStr : 'Unnamed Datasource';
+        this.setState({
+          datasource: dsStr
+        });
+      });
+
+    // props.logPageView( props.history.location );
```
Comment (jonahdlin) [inline src/client/features/view/index.js]: delete logpageview, logevent

File: src/server/graph-generation/metadataMapper.js [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+  var nodes = cyGraph.nodes;
+  cyGraph.nodes = processBioPax(biopax, nodes);
+
+  //Return Enhanced JSON
+  return cyGraph;
+}
+
+
+//Filters the given array to those which when passed into matcher return true
+Array.prototype.where = function (matcher) {
```
Comment (jonahdlin) [inline src/server/graph-generation/metadataMapper.js]: delete this

**Pull request merged**